KAS-2378 Upgrade ember simple auth

erikap opened the pull request:
Remove usages of ember-simple-auth mixins:
- Extend session service to replace ApplicationRouteMixin
- Use session.requireAuthentication instead of AuthenticatedRouteMixin
- Use session.prohibitAuthentication on login routes

Octanify and cleanup current-session service

erikap committed:
Remove usages of ember-simple-auth mixins

- Extend session service to replace ApplicationRouteMixin
- Use session.requireAuthentication instead of AuthenticatedRouteMixin
- Use session.prohibitAuthentication on login routes

Also discovered that app/pods/mock-login-route is not named
app/pods/mock-login on purpose to avoid collision with mock-login
component provided by the ember-mock-login addon.

erikap committed:
Remove default routes and ember-simple-auth mixins from styleguide

erikap committed:
Fix linting

erikap committed:
Show main header for users with role 'users'

erikap committed:
Update accountless-users template

To be more in line with auk-styleguide

erikap committed:
Add route-not-found template

To be more in line with auk-styleguide

erikap committed:
Refactor mock login route

To be more in line with prototype

erikap committed:
Refactor current sesssion to ES6 class

erikap committed:
Cleanup unused properties of current session service

erikap committed:
Fix test selector for mock-login

erikap committed:
Replace account group id config with URI constants

erikap committed:
Fix usage of currentSession.user in model-with-modifier

...and only update modification date if something actually changed

MikiDi commented on app/components/m-header.js:
```diff
@@ -26,8 +26,7 @@ export default class MHeader extends Component {
 
-    if (window.location.href.indexOf('https://kaleidos.vlaanderen.be') === 0
-      && this.currentSession.checkIsDeveloper()) {
+    if (window.location.href.indexOf('https://kaleidos.vlaanderen.be') === 0) {
```
Deze willen we waarschijnlijk in z'n geheel weg ... de eindgebruikers hebben weinig boodschap aan de pill `PROD` (?). +
- if -> else if
- window.location.hostname met straight `===` ipv `indexOf` ...

erikap replied:
Correct. 
Ik heb de logica nog wat verder opgekuist om op hostname te checken en met config-map te werken ipv `if-else`.

MikiDi commented on app/pods/application/route.js:
```diff
@@ -1,30 +1,50 @@
-  routeAfterAuthentication: 'agendas',
-  router: service(),
-  metrics: service(),
+function isSupportedBrowser() {
```
We gebruiken [elders](https://github.com/kanselarij-vlaanderen/kaleidos-frontend/blob/45b396bc483ce05cc3d45fe5d0cd0ec35a0a68e4/app/components/web-components/vl-rdfa-editor.js#L17) al https://github.com/willviles/ember-useragent ... misschien de moeite om mee te nemen?

erikap committed:
Cleanup rendering of environment pill in header

- No pill on production
- Simplify environment detection logic

erikap committed:
Use userAgent service to verify supported application browsers

erikap committed:
Merge branch 'development' into feature/KAS-2378-upgrade-ember-simple-auth

erikap committed:
Merge branch 'development' into feature/KAS-2378-upgrade-ember-simple-auth

erikap committed:
Merge branch 'development' into feature/KAS-2378-upgrade-ember-simple-auth

erikap committed:
Use constructor instead of init in route and controller

See also:
https://guides.emberjs.com/release/upgrading/current-edition/native-classes/#toc_constructor-instead-of-init

erikap committed:
Merge branch 'development' into feature/KAS-2378-upgrade-ember-simple-auth

MikiDi commented on app/services/session.js:
```diff
@@ -0,0 +1,25 @@
+
+  handleInvalidation() {
+    const logoutUrl = ENV.torii.providers['acmidm-oauth2'].logoutUrl;
+    if (logoutUrl.startsWith('http')) {
```
I this is a validation, it isn't super self-documenting. a `try ... catch` with [URL](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL) might be better suited?

erikap replied:
Indeed. Copied that snippet from lblod/loket :see_no_evil: 
Will change this. `try ... catch` is a better approach and will also handle the case if `logoutUrl` is not configured.

MikiDi commented on app/services/session.js:
```diff
@@ -0,0 +1,25 @@
+    if (logoutUrl.startsWith('http')) {
+      super.handleInvalidation(logoutUrl);
+    } else {
+      super.handleInvalidation('aanmelden');
```
Isn't an ember route (`login`) name expected here?

erikap replied:
That's what I tried first, but didn't work. Will double check to be sure and add a comment.

erikap replied:
Looking at the default implementation in ember-simple-auth (https://github.com/simplabs/ember-simple-auth/blob/80b9c152ff3e13e9d03b309a7a4195400a7533b2/packages/ember-simple-auth/addon/-internals/routing.js#L52-L60) it seems to expect a route name when FastBoot is enabled and a URL or path otherwise. I will refactor so we don't use `super.handleInvalidation` and just implement our own logic.

MikiDi commented on app/services/current-session.js:
```diff
@@ -1,171 +1,71 @@
-      ben: CONFIG.ben,
-      rafael: CONFIG.rafael,
-    };
+  get isOvrb() {
```
Future work, but I find these to be misleading. role-like naming would be better here imo, eg `canEditAgenda`, `canViewPublications`, `canEditPublications`, `canEditNewsItems`, ....

erikap replied:
Agree, but future work indeed.

erikap committed:
Merge branch 'development' into feature/KAS-2378-upgrade-ember-simple-auth

erikap committed:
Don't use ember-simple-auths default implementation on session invalidation
